### PR TITLE
GB-3774: Implement OpenAPI connector(s) in the SDK

### PR DESCRIPTION
**Pull request by pimeys**
# Description

Allows creating OpenAPI connectors in the SDK config generator.

# Type of change

- [ ] 💔 Breaking
- [x] 🚀 Feature
- [ ] 🐛 Fix
- [ ] 🛠️ Tooling
- [ ] 🔨 Refactoring
- [ ] 🧪 Test
- [ ] 📦 Dependency
- [x] 📖 Requires documentation update

### CODE REVIEW

File: packages/grafbase-sdk/README.md [hunk 1]
```diff
@@ -307,4 +307,36 @@ const user = g.model("User", {
...
+const openai = connector.OpenAPI({
+  schema: 'https://raw.githubusercontent.com/openai/openai-openapi/master/openapi.yaml'
+})
+
+const stripe = connector
+  .OpenAPI({
+    schema: 'https://raw.githubusercontent.com/stripe/openapi/master/openapi/spec3.json',
+    url: 'https://api.stripe.com'
+  })
+  .header('Authorization', 'Bearer {{ env.STRIPE_API_KEY }}')
```
Comment (fbjork) [inline packages/grafbase-sdk/README.md:334]: Should header be builder or another key inside OpenAPI?

File: packages/grafbase-sdk/README.md [hunk 2]
```diff
@@ -307,4 +307,36 @@ const user = g.model("User", {
...
+
+```typescript
+const openai = connector.OpenAPI({
+  schema: 'https://raw.githubusercontent.com/openai/openai-openapi/master/openapi.yaml'
+})
+
+const stripe = connector
+  .OpenAPI({
+    schema: 'https://raw.githubusercontent.com/stripe/openapi/master/openapi/spec3.json',
+    url: 'https://api.stripe.com'
```
Comment (fbjork) [inline packages/grafbase-sdk/README.md:332]: need introspectionHeaders here

File: packages/grafbase-sdk/README.md (hunk 1 above)
Comment (fbjork) [reply packages/grafbase-sdk/README.md:334]: Right, but then we have some attributes that are set via build and some that aren't. I like the consistency of having everything related to the connector inside the {} personally.

File: packages/grafbase-sdk/README.md [hunk 3]
```diff
@@ -307,4 +307,41 @@ const user = g.model("User", {
...
+
+```typescript
+const openai = connector.OpenAPI({
+  schema: 'https://raw.githubusercontent.com/openai/openai-openapi/master/openapi.yaml'
+})
+
+const stripe = connector
+  .OpenAPI({
+    schema: 'https://raw.githubusercontent.com/stripe/openapi/master/openapi/spec3.json',
+    url: 'https://api.stripe.com',
```
[the remark this answers is not in the document]
Comment (pimeys) [reply packages/grafbase-sdk/README.md:332]: It's an optional parameter here. You don't have to define it.

File: packages/grafbase-sdk/src/openapi.ts [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+  }
+}
+
+export type HeaderGenerator = (headers: Headers) => any
+
+export interface OpenAPIParams {
+  schema: string
+  url?: string
+  headers?: HeaderGenerator
+}
```
Comment (obmarg) [inline packages/grafbase-sdk/src/openapi.ts:51]: We've also got a [transforms parameter](https://github.com/grafbase/api/blob/main/common/parser/src/rules/openapi_directive.rs#L19-L27)

**Review by fbjork — APPROVED** [review]
nice work. added a minor comment.

**Pull request merged**